### PR TITLE
Final Sprint2 PR

### PR DESCRIPTION
**Pull request by crobinson3209**

### CODE REVIEW

File: server/src/webserver.rs [hunk 1]
```diff
@@ -927,7 +955,37 @@ mod tests {
         let llc = Data::new(compile_lexers(LANGUAGE_LEXER_ARR)); 
         let (file_contents, _) = save_source_as_string(actix_web::web::Json(test_source_file), llc);
         assert_eq!(file_contents, "# Test");
+    
+        
+        // <p><strong>Pass a block comment</strong></p>
+        let test_source_file = ClientSourceFile{
+            metadata: SourceFileMetadata {
+                mode: "python".to_string()
```
Comment (bjones1) [inline server/src/webserver.rs:963]: Python doesn't have block comments -- use C/C++/etc. instead.

File: server/src/webserver.rs [hunk 2]
```diff
@@ -927,7 +955,37 @@ mod tests {
...
+        }; 
+        let llc = Data::new(compile_lexers(LANGUAGE_LEXER_ARR)); 
+        let (file_contents, _) = save_source_as_string(actix_web::web::Json(test_source_file), llc);
+        assert_eq!(file_contents, "This is a block comment");
+    
+    
+        // <p><strong>Pass an inline comment</strong></p>
+        let test_source_file = ClientSourceFile{
+            metadata: SourceFileMetadata {
+                mode: "python".to_string()
```
Comment (bjones1) [inline server/src/webserver.rs:977]: Python's inline comment is `#`, not `//`.

File: server/src/webserver.rs [hunk 3]
```diff
@@ -927,7 +955,37 @@ mod tests {
...
+            },
+            code_doc_block_arr: vec![
+                ("".to_string(),Some("".to_string()),"This is some code // with an inline comment".to_string()),
+            ] 
+        }; 
+        let llc = Data::new(compile_lexers(LANGUAGE_LEXER_ARR)); 
+        let (file_contents, _) = save_source_as_string(actix_web::web::Json(test_source_file), llc);
+        assert_eq!(file_contents, "This is some code // with an inline comment");
     }
+
```
Comment (bjones1) [inline server/src/webserver.rs:987]: Run `cargo fmt` and `cargo clippy` to clean up whitespace / syntax.

**Review by bjones1 — CHANGES_REQUESTED** [review]

File: server/src/webserver.rs (hunk 2 above)
Comment (crobinson3209) [reply server/src/webserver.rs]: made this correction and also added a c/c++ test for inline comments

File: server/src/webserver.rs (hunk 1 above)
Comment (crobinson3209) [reply server/src/webserver.rs]: made this correction

File: server/src/webserver.rs [hunk 4]
```diff
@@ -915,7 +915,35 @@ mod tests {
...
+        let (file_contents, _) = save_source_as_string(actix_web::web::Json(test_source_file), llc);
+        assert_eq!(file_contents, "Test");
+//
+        // <p><strong>Pass only an indent</strong></p>
+        let test_source_file = ClientSourceFile{
+            metadata: SourceFileMetadata {
+                mode: "python".to_string()
+            },
+            code_doc_block_arr: vec![
+                (" ".to_string(),Some("".to_string()),"".to_string()),
```
[the remark this answers is not in the document]
Comment (crobinson3209) [reply server/src/webserver.rs]: made this correction and also implemented it for C/C++

File: server/src/webserver.rs (hunk 3 above)
Comment (samcarley) [reply server/src/webserver.rs]: Done!

**Comment by samcarley** [comment]
Final PR is now ready. Had to fix a few errors, edit a test. Also cleaned up the whitespace.

**Pull request merged**